registration/views.py

```diff
@@ -122,11 +123,17 @@ def setup_external_source_session(request):
         try:
             clean_widget_session(request.session)
             clean_email_source_session(request.session)
-            cobrand_company = cobrand.models.CobrandCompany.objects.get(uuid=cobrand_id)
+
+            company_name = cache.get(cobrand_id)
+            if not company_name:
+                cobrand_company = cobrand.models.CobrandCompany.objects.get(uuid=cobrand_id)
+                company_name = cobrand_company.company_name
+                cache.set(cobrand_id, company_name)
```

Comment:
@dana Should we set a timeout knowing that we already have a default timeout set to 1 day in base.py for CACHES?